includes/api/class-wc-admin-rest-products-controller.php

```diff
@@ -89,50 +98,68 @@ protected function prepare_objects_query( $request ) {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_items( $request ) {
-		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10, 2 );
-		add_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10, 2 );
-		add_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_product_search_group_by' ), 10, 2 );
+		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_filter' ), 10, 2 );
+		add_filter( 'posts_join', array( __CLASS__, 'add_wp_query_join' ), 10, 2 );
+		add_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_group_by' ), 10, 2 );
 		$response = parent::get_items( $request );
-		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10 );
-		remove_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10 );
-		remove_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_product_search_group_by' ), 10 );
+		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_filter' ), 10 );
+		remove_filter( 'posts_join', array( __CLASS__, 'add_wp_query_join' ), 10 );
+		remove_filter( 'posts_groupby', array( __CLASS__, 'add_wp_query_group_by' ), 10 );
 		return $response;
 	}
 
 	/**
-	 * Allow searching by product name or sku in WP Query
+	 * Add in conditional search filters for products.
 	 *
 	 * @param string $where Where clause used to search posts.
 	 * @param object $wp_query WP_Query object.
 	 * @return string
 	 */
-	public static function add_wp_query_product_search_filter( $where, $wp_query ) {
+	public static function add_wp_query_filter( $where, $wp_query ) {
 		global $wpdb;
 
-		$search = trim( $wp_query->get( 'search' ) );
+		$search = $wp_query->get( 'search' );
 		if ( $search ) {
 			$search = $wpdb->esc_like( $search );
 			$search = "'%" . $search . "%'";
-			$where .= ' AND (' . $wpdb->posts . '.post_title LIKE ' . $search;
+			$where .= " AND ({$wpdb->posts}.post_title LIKE {$search}";
 			$where .= wc_product_sku_enabled() ? ' OR ps_post_meta.meta_key = "_sku" AND ps_post_meta.meta_value LIKE ' . $search . ')' : ')';
 		}
 
+		if ( $wp_query->get( 'low_in_stock' ) ) {
+			$low_stock_amount = get_option( 'woocommerce_notify_low_stock_amount', 2 );
```

Comment:
We need to either SQL escape this or parse it as an integer (`get_option` is filterable).

Reply:
Updated in https://github.com/woocommerce/woocommerce-admin/pull/1866/commits/e466d4029fda1ef2528af988060c2391c3706cd4